crates/bevy_ecs/src/schedule/executor/multi_threaded.rs

```diff
@@ -516,132 +600,84 @@ impl MultiThreadedExecutor {
     ///   used by the specified system.
     /// - `update_archetype_component_access` must have been called with `world`
     ///   on the system associated with `system_index`.
-    unsafe fn spawn_system_task<'scope>(
-        &mut self,
-        scope: &Scope<'_, 'scope, ()>,
-        system_index: usize,
-        systems: &'scope [SyncUnsafeCell<BoxedSystem>],
-        world: UnsafeWorldCell<'scope>,
-    ) {
+    unsafe fn spawn_system_task(&mut self, context: &Context<'_, '_, '_>, system_index: usize) {
         // SAFETY: this system is not running, no other reference exists
-        let system = unsafe { &mut *systems[system_index].get() };
-        let sender = self.sender.clone();
-        let panic_payload = self.panic_payload.clone();
+        let system = unsafe { &mut *context.environment.systems[system_index].get() };
+        // Move the full context object into the new future.
+        let context = *context;
+
+        let system_meta = &self.system_task_metadata[system_index];
+
+        #[cfg(feature = "trace")]
+        let system_span = system_meta.system_task_span.clone();
```

Comment:
Oh, I see, I made this span less useful by pulling the sending of the completion message out of it.  Sorry; I was trying to exclude `tick_executor()` and I had put those in the same function.  

The existing code was doing a non-blocking `try_send` on the channel, which calls `ConcurrentQueue::push` and then does some bookkeeping.  So the thread wouldn't have parked before and won't park now.  It looks like [there is a busy-wait in `push`](https://github.com/smol-rs/concurrent-queue/blob/93ee058b7f07b2346fdd55af6a033165239ef477/src/bounded.rs#L150), though, so that may be slow under contention.  

Do you want any changes here?  I could take the `tick_executor()` call out of `system_completed()` and then put `system_completed()` back in the tracing span.  Or I could remove the span.

Reply:
I think it's fine to leave as is for now. If we're not seeing much different here between the system run time we should probably remove it later.